x-pack/plugins/apm/public/hooks/use_kuery_with_mobile_filters.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import {
+  HOST_OS_VERSION,
+  DEVICE_MODEL_NAME,
+  NETWORK_CONNECTION_TYPE,
+  SERVICE_VERSION,
+} from '../../common/es_fields/apm';
+import { paramQuery } from '../../common/utils/param_query';
+
+export function useKueryWithMobileFilters({
+  device,
+  osVersion,
+  appVersion,
+  netConnectionType,
+  kuery,
+}: {
+  device?: string;
+  osVersion?: string;
+  appVersion?: string;
+  netConnectionType?: string;
+  kuery?: string;
```

Comment:
I think we should at least consider making these required (at the route level, and then have it cascade down). Otherwise there's a chance of us forgetting about including the parameters when linking to other pages.